mcx/trace.py

```diff
@@ -1,31 +1,44 @@
-class Trace:
+from typing import Dict
+import mcx
+from arviz import InferenceData
+from arviz.data.base import dict_to_dataset
+
+
+__all__ = ["Trace"]
+
+
+class Trace(InferenceData):
     """Trace contains the data generated during inference: samples,
     divergences, values of diagnostics, etc.
-    """
 
-    def __init__(self):
-        self.posterior = None
-        self.posterior_predictive = None
-        self.sample_stats = None
-        self.log_likelihood = None
-        self.posterior_predictive = None
-        pass
+    The class is a thin wrapper around ArviZ's InferenceData, it is an
+    interface between the chains produced by the samplers and ArviZ.
+    """
 
-    @classmethod
-    def from_dict(cls, dictionary) -> "Trace":
-        pass
+    def __init__(self, *, posterior: Dict = None, coords=None, dims=None):
+        """Build a Trace object from MCX data.
 
-    def to_dict(self) -> dict:
-        """Returns the trace in a dictionary format that ArviZ can understand.
-        The resulting dict can be persisted on disk in the json and yaml format
-        to be loaded later.
+        Parameters
+        ----------
+        posterior
+            Posterior samples from a model. The dictionary maps the variables
+            names to their posterior samples with shape (n_chains, num_samples, var_shape).
         """
-        return {
-            "posterior": self.posterior,
-            "posterior_predictive": self.posterior_predictive,
-            "sample_stats": self.sample_stats,
-            "log_likelihood": self.log_likelihood,
+        samples_dataset = dict_to_dataset(
+            data=posterior["samples"], library=mcx, coords=coords, dims=dims
+        )
+
+        sample_stats_dict = {
+            "lp": posterior["potential_energy"],
+            "mean_tree_accept": posterior["acceptance_probability"],
```

Comment:
```suggestion
            "acceptance_rate": posterior["acceptance_probability"],
```

We are working on a naming convention for sample stats. Right now only lp and diverging are enforced, the other stats have different names depending on the sampling library. 

Currently ArviZ only uses `diverging` and `energy` in its plots, but with time more quantities will probably be used, probably even more now with [arviz_dashboard](https://github.com/arviz-devs/arviz_dashboard).

Reply:
Perfect timing! I was going to ask you about this since many names are NUTS-specific, and I have only implemented HMC so far.